src/OpenTracingActiveRecordBehavior.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Websupport\OpenTracing;
+
+use CEvent;
+use CModelEvent;
+use OpenTracing\Scope;
+
+/**
+ * Class OpenTracingActiveRecordBehavior
+ * @package Websupport\OpenTracing
+ *
+ * @method \CActiveRecord getOwner()
+ */
+class OpenTracingActiveRecordBehavior extends \CActiveRecordBehavior
+{
+    /** @var string */
+    public $opentracing = 'opentracing';
+
+    /**
+     * Whether we should hook to before and after find methods.
+     * Be careful with this option. There is option, where you can disable calling afterFind method
+     * directly in the code.
+     * @var bool
+     */
+    public $traceFind = false;
+
+    /**
+     * Whether we should hook to before and after save methods.
+     * @var bool
+     */
+    public $traceSave = true;
+
+    /**
+     * Whether we should hook to before and after delete methods.
+     * @var bool
+     */
+    public $traceDelete = true;
+
+    /** @var Scope */
+    private $activeScope = null;
+
+    /**
+     * Responds to {@link CActiveRecord::onBeforeSave} event.
+     * Override this method and make it public if you want to handle the corresponding
+     * event of the {@link CBehavior::owner owner}.
+     * You may set {@link CModelEvent::isValid} to be false to quit the saving process.
+     * @param CModelEvent $event event parameter
+     * @throws
+     */
+    public function beforeSave($event)
+    {
+        if(!$this->traceSave && !\Yii::app()->hasComponent($this->opentracing)) {
+            return;
+        }
+
+        $this->activeScope = \Yii::app()->hasComponent($this->opentracing)->startActiveSpan(
+            $this->spanName($this->getOwner()->getIsNewRecord() ? 'INSERT' : 'UPDATE'),
+            $this->spanOptionsFromActiveRecordClass()
+        );
+        $this->activeScope->getSpan()->log([
+            'attributes' => $this->getOwner()->getAttributes()
+        ]);
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onAfterSave} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function afterSave($event)
+    {
+        if(!$this->traceSave && $this->activeScope) {
```

Comment:
`$this->activeScope` is valid also when previous active scope was closed

Reply:
fixed